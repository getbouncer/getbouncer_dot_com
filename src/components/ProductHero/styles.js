import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const InnerContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-bottom: 90px;
  padding: 60px 0 0;

  @media ${breakpoints.smartphone} {
    display: block;
    margin-bottom: 70px;
  }
`

export const Image = styled.img`
  width: 50%;

  @media ${breakpoints.smartphone} {
    margin-bottom: 40px;
    width: 100%;
  }
`

export const ProductLogo = styled.img`
  height: 36px;
  margin-bottom: 18px;
`

export const Headline = styled.div`
  max-width: 485px;

  strong {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
`

export const CtaContainer = styled.div`
  margin-top: 60px;
`
