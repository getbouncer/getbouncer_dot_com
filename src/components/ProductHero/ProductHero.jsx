import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import CtaButton from '~components/CtaButton'
import { InnerContainer, Image, ProductLogo, Headline, CtaContainer } from './styles'

const ProductHero = ({ data }) => {
  const {
    product_cta: productCta,
    product_cta_link: productCtaLink,
    product_headline: productHeadline,
    product_image: productImage,
    product_logo: productLogo,
  } = data

  return (
    <Container>
      <InnerContainer>
        <Image src={productImage.url} alt={productImage.alt} />
        <div>
          <ProductLogo src={productLogo.url} alt={productLogo.alt} />
          <Headline dangerouslySetInnerHTML={{ __html: productHeadline.html }} />
          {productCta && (
            <CtaContainer>
              <CtaButton isInternal to={productCtaLink} text={productCta} />
            </CtaContainer>
          )}
        </div>
      </InnerContainer>
    </Container>
  )
}

ProductHero.propTypes = {
  data: PropTypes.shape().isRequired,
}

export default ProductHero
