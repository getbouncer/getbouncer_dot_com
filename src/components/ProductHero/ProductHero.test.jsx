import React from 'react'
import renderer from 'react-test-renderer'
import ProductHero from './ProductHero'

/* eslint-disable no-undef */

describe('ProductHero', () => {
  it('renders correctly', () => {
    const data = {
      product_name: {
        text: 'product',
      },
      product_logo: {
        alt: 'product logo',
        url: 'https://www.example.com/product_logo.svg',
      },
      product_image: {
        alt: 'product image',
        url: 'https://www.example.com/product_image.png',
      },
      product_headline: {
        html: '<h1>product headline</h1>',
        text: 'product headline',
      },
      product_cta: 'product cta',
      show_get_started: true,
    }
    const tree = renderer
      .create(<ProductHero data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
