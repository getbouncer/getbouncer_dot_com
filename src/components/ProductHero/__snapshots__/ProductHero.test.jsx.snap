// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHero renders correctly 1`] = `
<div
  className="css-1fywgw4"
>
  <div
    className="css-1gv4vrc"
  >
    <img
      alt="product image"
      className="css-k5imqe"
      src="https://www.example.com/product_image.png"
    />
    <div>
      <img
        alt="product logo"
        className="css-795m1v"
        src="https://www.example.com/product_logo.svg"
      />
      <div
        className="css-s5mp0m"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h1>product headline</h1>",
          }
        }
      />
      <div
        className="css-12sqe98"
      >
        <a
          className="css-e1lh64"
          href="/contact"
          inverted={false}
        >
          product cta
        </a>
      </div>
    </div>
  </div>
</div>
`;
